src/node/mod.rs

```diff
@@ -643,6 +643,33 @@ impl Node {
     ) -> anyhow::Result<impl Stream<Item = BlockNotification> + Send + Sync + Unpin + 'static> {
         self.rpc_handle.subscribe_new_blocks().await.context("Failed to subscribe to new blocks")
     }
+
+    pub(crate) async fn get_events_inner(
+        &self,
+        block: Option<Hash>,
+    ) -> anyhow::Result<Vec<frame_system::EventRecord<Event, Hash>>> {
+        match self
+            .rpc_handle
+            .get_storage(StorageKey::events(), block)
+            .await
+            .context("Failed to get events from storage")?
+        {
+            Some(sp_storage::StorageData(events)) =>
+                parity_scale_codec::DecodeAll::decode_all(&mut events.as_ref())
+                    .context("Failed to decode events"),
+            None => Ok(vec![]),
+        }
+    }
+
+    /// Get events at some block or at tip of the chain
+    pub async fn get_events(&self, block: Option<Hash>) -> anyhow::Result<Vec<Event>> {
+        Ok(self
+            .get_events_inner(block)
+            .await?
+            .into_iter()
+            .map(|event_record| event_record.event)
+            .collect())
+    }
```

Comment:
is there a reason you don’t wanna inline `get_events_inner` into `get_events?` seems like the signature is the same. And I don't see any branching factor?